src/types/ProfileUpdater.cpp

```diff
@@ -0,0 +1,352 @@
+    altPerf_model.changeMode("alt");
+    auto alternative = altPerf_model.operator[](alt.getName());
+    std::vector<std::vector<Perf>> pt = model.profiles.getPerformanceTable();
+    Category cat_new = model.categoryAssignment(alternative, pt);
+    std::string aa_new = cat_new.getCategoryId();
+
+    // Update alternative assignment
+    altPerf_model.setAlternativeAssignment(alt.getName(), cat_new);
+
+    std::cout << alt.getName() << "=> aa_data: " << aa_data
+              << " aa_old: " << aa_old << " aa_new: " << aa_new << std::endl;
+    std::cout << "c: " << c << std::endl;
+
+    // Update good assignment count
+    if (aa_old == aa_new) {
+      std::cout << "no change" << std::endl;
+      break;
+    } else if (aa_old == aa_data) {
+      good = good - 1;
+      std::cout << "one less :( " << std::endl;
+    } else if (aa_new == aa_data) {
+      good = good + 1;
+      std::cout << "one more :D " << std::endl;
+    }
+  }
+}
+
+void ProfileUpdater::optimizeProfile(
+    std::vector<Perf> &prof, Category &cat_below, Category &cat_above,
+    MRSortModel &model,
+    std::unordered_map<std::string, std::unordered_map<std::string, float>> &ct,
+    AlternativesPerformance &altPerf_model) {
+
+  std::pair<std::vector<Perf>, std::vector<Perf>> below_above =
+      model.profiles.getBelowAndAboveProfile(prof[0].getName());
+  std::vector<Perf> prof_below = below_above.first;
+  std::vector<Perf> prof_above = below_above.second;
+
+  for (Criterion crit : model.criteria.getCriterionVect()) {
+    Perf b = getPerfOfCrit(prof, crit.getId());
+    Perf b_below = getPerfOfCrit(prof_below, crit.getId());
+    Perf b_above = getPerfOfCrit(prof_above, crit.getId());
+
+    std::unordered_map<std::string, float> ct_prof = ct[b.getName()];
+
+    std::unordered_map<float, float> below_des = this->computeBelowDesirability(
+        model, crit.getId(), b, b_below, cat_below, cat_above, ct_prof,
+        altPerf_model);
+    std::unordered_map<float, float> above_des = this->computeAboveDesirability(
+        model, crit.getId(), b, b_above, cat_below, cat_above, ct_prof,
+        altPerf_model);
+    std::unordered_map<float, float> desirability = below_des;
+    desirability.insert(above_des.begin(), above_des.end());
+
+    float key_max = this->chooseMaxDesirability(desirability, b);
+
+    float r = getRandomUniformFloat();
+    if (r <= key_max) {
+      Perf b_new = Perf(b);
+      b_new.setValue(key_max);
+      this->updateTables(model, crit.getId(), b, b_new, ct, good_,
+                         altPerf_model);
+    }
+    std::cout << crit << std::endl;
+    std::cout << desirability << std::endl;
+    std::cout << key_max << std::endl;
+  }
+}
+
+float ProfileUpdater::optimize(
+    MRSortModel &model,
+    std::unordered_map<std::string, std::unordered_map<std::string, float>> &ct,
+    AlternativesPerformance &altPerf_model) {
+  int i = 0;
+  for (std::vector<Perf> profile : model.profiles.getPerformanceTable()) {
```

Comment:
In this function i is always set to 0 ?

Reply:
Maybe you made test for only 2 categorie, idk ?

Reply:
lol yes my bad!